cmd/ddev/cmd/config.go

```diff
@@ -56,12 +58,33 @@ var ConfigCommand = &cobra.Command{
 		c.Name = siteName
 		c.Docroot = docrootRelPath
 
+		// If they have not given us any flags, we prompt for info. Otherwise, we assume they're in control.
 		if siteName == "" && docrootRelPath == "" && pantheonEnvironment == "" && appType == "" {
 			err = c.PromptForConfig()
 			if err != nil {
 				util.Failed("There was a problem configuring your application: %v\n", err)
 			}
-		} else {
+		} else { // In this case we have to validate the provided items, or set to sane defaults
+			// siteName gets set to basename if not provided
+			if siteName == "" {
+				pwd, err := os.Getwd()
+				util.CheckErr(err)
+				siteName = path.Base(pwd)
+			}
+			// docrootRelPath must exist
+			if _, err = os.Stat(docrootRelPath); docrootRelPath != "" && os.IsNotExist(err) {
+				util.Failed("The docroot provided (%v) does not exist", docrootRelPath)
+			}
+			// pantheonEnvironment must be appropriate, and can only be used with pantheon provider.
+			if provider != "pantheon" && pantheonEnvironment != "" {
+				util.Failed("--pantheon-environment can only be used with pantheon provider, for example ddev config pantheon --pantheon-environment=dev --docroot=docroot")
+			}
+			if pantheonEnvironment != "" && pantheonEnvironment != "dev" && pantheonEnvironment != "test" && pantheonEnvironment != "prod" {
```

Comment:
Small nit: in multidev environments a different value is completely valid. The pantheon plugin will validate this and the sitename if you use `provider.Validate()` I believe.